src/pdf/ShoppingList/dataPreparation.ts

```diff
@@ -0,0 +1,247 @@
+    collectionSite: string;
+}
+
+export interface ClientSummary {
+    name: string;
+    contact: string;
+    address: string;
+    extraInformation: string;
+}
+
+export interface HouseholdSummary extends BlockProps {
+    householdSize: string;
+    genderBreakdown: string;
+    numberOfBabies: string;
+    ageAndGenderOfChildren: string;
+}
+
+export interface RequirementSummary extends BlockProps {
+    feminineProductsRequired: string;
+    babyProductsRequired: string;
+    petFoodRequired: string;
+    dietaryRequirements: string;
+    otherItems: string;
+}
+
+export interface Item {
+    description: string;
+    quantity: string;
+    notes: string;
+}
+
+const formatDate = (dateString: string | null): string => {
+    if (!dateString) {
+        return "";
+    }
+    return new Date(dateString).toLocaleString("en-GB", {
+        year: "numeric",
+        month: "numeric",
+        day: "numeric",
+        hour: "numeric",
+        minute: "numeric",
+    });
+};
+
+const prepareParcelInfo = async (parcelID: string): Promise<ParcelInfoAndClientID> => {
+    const fetchedData = await fetchParcels(parcelID);
+    const parcelInfo: ParcelInfo = {
+        voucherNumber: fetchedData.voucher_number ?? "",
+        packingDate: formatDate(fetchedData.packing_datetime),
+        collectionDate: formatDate(fetchedData.collection_datetime),
+        collectionSite: fetchedData.collection_centre ?? "",
+    };
+
+    if (parcelInfo.collectionSite === "Delivery") {
+        parcelInfo.collectionSite = "N/A - Delivery";
+    }
+
+    return { parcelInfo: parcelInfo, clientID: fetchedData.client_id };
+};
+
+const getClientAndFamilyData = async (clientID: string): Promise<ClientDataAndFamilyData> => {
+    const clientData = await fetchClients(clientID);
+    const familyData = await fetchFamilies(clientData.family_id);
+
+    return { clientData: clientData, familyData: familyData };
+};
+
+const processExtraInformation = (original: string): NappySizeAndExtraInformation => {
+    if (original.startsWith("Nappy Size: ")) {
+        const [nappySize, extraInformation] = original.split(", Extra Information: ");
+        return { nappySize: `(${nappySize})`, extraInformation: extraInformation };
+    }
+    return { nappySize: "", extraInformation: original };
+};
+
+const prepareClientSummary = (clientData: Schema["clients"]): ClientSummary => {
+    const {
+        address_1,
+        address_2,
+        address_county,
+        address_postcode,
+        address_town,
+        full_name,
+        phone_number,
+    } = clientData;
+    const address = [address_1, address_2, address_county, address_postcode, address_town]
+        .filter((value) => value !== "")
+        .join("\n");
+
+    return {
+        name: full_name,
+        contact: phone_number,
+        address: address,
+        extraInformation: "",
+    };
+};
+
+const getChild = (child: Person): string => {
+    return `${child.age} ${child.gender[0].toUpperCase()}`;
```

Comment:
does `13 O` for 'other' make sense?

Reply:
I agree that O on its own might not be useful, will change O to N/A but will keep M and F as is.